tests/test_functions.py

```diff
@@ -52,6 +58,17 @@ def test_functions():
     assert dill.loads(dumped_func_d)(1, 2, 3) == 6
     assert dill.loads(dumped_func_d)(1, 2, d2=3) == 6
 
+    empty_cell = make_empty_cell()
```

Comment:
It seems that `empty_cell`, `copy_cell`, and `make_empty_cell` are poorly named as `make_empty_cell` actually creates a function where the closure has an empty cell.  These could be renamed.

Reply:
LGTM